sdk/nodejs/automation/stack.ts

```diff
@@ -454,9 +454,12 @@ Event: ${line}\n${e.toString()}`);
         const kind = this.workspace.program ? execKind.inline : execKind.local;
         args.push("--exec-kind", kind);
 
-        const refPromise = this.runPulumiCmd(args, opts?.onOutput);
-        const [refResult, logResult] = await Promise.all([refPromise, logPromise]);
-        await cleanUp(logFile, logResult);
+        let refResult: CommandResult;
+        try {
+            refResult = await this.runPulumiCmd(args, opts?.onOutput);
+        } finally {
+            await cleanUp(logFile, await logPromise);
```

Comment:
Could the `logPromise` throw an exception as well?  Or can we assume that that's always going to run fine?

Reply:
I mirrored what we have for `up` and `preview` https://github.com/pulumi/pulumi/blob/3eef56baeb4a340efd0e2b1813821b0999739396/sdk/nodejs/automation/stack.ts#L261, so I think it's mostly safe.